src/scheduler/Scheduler.java

```diff
@@ -109,7 +111,7 @@ public void run() {
             }
         }
 
-        logger.addMessage("AVERAGE_EXCHANGES", changes);
+        logger.addMessage("AVERAGE_EXCHANGES", changes / numOfProcess);
```

Comment:
If you look at file gave by professor you could see that the correct number should be only `changes` (as it was), so this division is wrong